keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/StepsizeTest.java

```diff
@@ -0,0 +1,103 @@
+package io.improbable.keanu.algorithms.mcmc;
+
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ProbabilityCalculator;
+import io.improbable.keanu.vertices.VertexId;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.LogProbGradientCalculator;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class StepsizeTest {
+
+    private KeanuRandom random = new KeanuRandom(1);
+
+    @Test
+    public void canFindSmallStartingStepsizeForSmallSpace() {
+        DoubleVertex vertex = new GaussianVertex(0, 0.05);
+        List<DoubleVertex> vertices = Arrays.asList(vertex);
+        BayesianNetwork bayesianNetwork = new BayesianNetwork(vertex.getConnectedGraph());
+
+        VertexId vertexId = vertex.getId();
+
+        LogProbGradientCalculator logProbGradientCalculator = new LogProbGradientCalculator(bayesianNetwork.getLatentOrObservedVertices(), vertices);
+        vertex.setValue(DoubleTensor.scalar(1.));
+        Map<VertexId, DoubleTensor> position = Collections.singletonMap(vertexId, vertex.getValue());
+        Map<VertexId, DoubleTensor> gradient = logProbGradientCalculator.getJointLogProbGradientWrtLatents();
+
+        double startingStepsize = Stepsize.findStartingStepSize(
+            position,
+            gradient,
+            Arrays.asList(vertex),
+            bayesianNetwork.getLatentVertices(),
+            logProbGradientCalculator,
+            ProbabilityCalculator.calculateLogProbFor(vertices),
+            random
+        );
+
+        double startingEpsilon = 1.0;
+        Assert.assertTrue(startingStepsize < startingEpsilon);
+    }
+
+    @Test
+    public void canFindLargeStartingStepsizeForLargeSpace() {
```

Comment:
lots of code duplication that could be refactored between `canFindSmallStartingStepsizeForSmallSpace` and `canFindLargeStartingStepsizeForLargeSpace`